code/noRAM/app/src/main/java/com/example/noram/MainActivity.java

```diff
@@ -42,6 +43,9 @@ public class MainActivity extends AppCompatActivity {
     public static Attendee attendee = null;
     public static Organizer organizer = null;
     private Button adminButton;
+    public static PushNotificationService pushService = new PushNotificationService();
+    public static MainActivity mn;
```

Comment:
main activity has an instance of itself? that seems odd. I feel like it could be better to have a getInstance() method or something like that?

Reply:
it needs to be static this is how I saw everyone do it

Reply:
a static method getInstance() wouldn't work?

If its the only way to do it then no worries, I just find this to be a confusing way to do this, since we have MainActivity has-a MainActivity which is strange

Reply:
ill try it

Reply:
I cant get it to work

Reply:
in that case we won't worry about it - small chance ill give it a crack at some point but not a huge deal